src/components/license-activation/LicenseActivation.jsx

```diff
@@ -1,27 +1,33 @@
-import React, { useContext } from 'react';
-import { Redirect, useParams, useLocation } from 'react-router-dom';
+import React, { useContext, useEffect, useState } from 'react';
+import { Redirect, useLocation } from 'react-router-dom';
 import { Helmet } from 'react-helmet';
 import { AppContext } from '@edx/frontend-platform/react';
 import { Alert, Container } from '@edx/paragon';
 
 import { LoadingSpinner } from '../loading-spinner';
-import { useLicenseActivation } from './data/hooks';
 
 import { useRenderContactHelpText } from '../../utils/hooks';
+import LicenseActivationErrorAlert from './LicenseActivationErrorAlert';
+import { UserSubsidyContext } from '../enterprise-user-subsidy/UserSubsidy';
 
 export const LOADING_MESSAGE = 'Your enterprise license is being activated! You will be automatically redirected to your organization\'s learner portal shortly.';
 
-export default function LicenseActivation() {
-  const { activationKey } = useParams();
+const LicenseActivation = () => {
   const { enterpriseConfig } = useContext(AppContext);
   const renderContactHelpText = useRenderContactHelpText(enterpriseConfig);
   const location = useLocation();
+  const { activateUserLicense } = useContext(UserSubsidyContext);
+  const [activationSuccess, setActivationSuccess] = useState();
 
-  const [activationSuccess, activationError] = useLicenseActivation({
-    activationKey,
-    enterpriseUUID: enterpriseConfig.uuid,
-    autoActivated: !!location.state?.from,
-  });
+  useEffect(() => {
+    const activateLicense = async () => {
+      const autoActivated = !!location.state?.from;
+      const isLicenseActivated = await activateUserLicense(autoActivated);
```

Comment:
[nit/curious] Interested in how we're relying on a boolean return value when calling `activateUserLicense` to gauge whether it's activated or not, versus relying on the `subscriptionLicense` from `UserSubsidyContext` which is presumably updated after a successful activation, e.g. something along the lines of:

```jsx
const { subscriptionLicense, activateUserLicense } = useContext(UserSubsidyContext);
const fromLocation = location.state?.from; // see below PR comment on linting for why this change is here
useEffect(() => {
  const activateLicense = async () => {
    const autoActivated = !!fromLocation;
    await activateUserLicense(autoActivated);
  };
  activateLicense();
}, [activateUserLicense, fromLocation]);

useEffect(() => {
  if (subscriptionLicense.status === LICENSE_STATUS.ACTIVATED) {
    setActivationSuccess(true);
  }
}, [subscriptionLicense.status]);
```

Just curious to hear your thoughts... I think either approach is fine!

Reply:
switched to throwing error in `activateLicense`, we don't need the useEffect hook since we're checking for error.